nats/js/object_store.py

```diff
@@ -0,0 +1,328 @@
+        def size(self) -> int:
+            """
+            size is the combined size of all data in the bucket including metadata, in bytes.
+            """
+            return self.stream_info.state.bytes
+
+    def __init__(
+        self,
+        name: str,
+        stream: str,
+        js: "JetStreamContext",
+    ) -> None:
+        self._name = name
+        self._stream = stream
+        self._js = js
+
+    def __sanitize_name(self, name: str) -> str:
+        name = name.replace(".", "_")
+        return name.replace(" ", "_")
+
+    async def get_info(self, name: str) -> api.ObjectInfo:
+        """
+        GetInfo will retrieve the current information for the object.
+        """
+        obj = self.__sanitize_name(name)
+
+        if not key_valid(obj):
+            raise InvalidObjectNameError
+
+        meta = OBJ_META_PRE_TEMPLATE.format(bucket=self._name, obj=obj)
+        stream = OBJ_STREAM_TEMPLATE.format(bucket=self._name)
+
+        msg = await self._js.get_last_msg(stream, meta)
+
+        data = None
+        if msg.data:
+            data = base64.b64decode(msg.data)
+
+        try:
+            info = api.ObjectInfo.from_response(json.loads(data))
+        except Exception as e:
+            raise BadObjectMetaError from e
+
+        # TODO: Uncomment below when time is added to RawStreamMsg.
+        # info.mtime = m.time
+
+        return info
+
+    async def get(self, name: str) -> ObjectResult:
+        """
+        Get will pull the object from the underlying stream.
+        """
+        obj = self.__sanitize_name(name)
+
+        if not key_valid(obj):
+            raise InvalidObjectNameError
+
+        # Grab meta info.
+        info = await self.get_info(obj)
+
+        if info.nuid is None or info.nuid == "":
+            raise BadObjectMetaError
+
+        # Check for object links.If single objects we do a pass through.
+        if info.is_link():
+            if info.options.link.name is None or info.options.link.name == "":
+                raise LinkIsABucketError
+            lobs = await self._js.object_store(info.options.link.bucket)
+            return await lobs.get(info.options.link.name)
+
+        result = self.ObjectResult(info=info)
+
+        if info.size == 0:
+            return result
+
+        chunk_subj = OBJ_CHUNKS_PRE_TEMPLATE.format(
+            bucket=self._name, obj=info.nuid
+        )
+        sub = await self._js.subscribe(
+            subject=chunk_subj, ordered_consumer=True
+        )
+
+        h = sha256()
+
+        async for msg in sub._message_iterator:
+            tokens = msg._get_metadata_fields(msg.reply)
+
+            result.data += msg.data
+            h.update(msg.data)
+
+            # Check if we are done.
+            if tokens[8] == OBJ_NO_PENDING:
+                await sub.unsubscribe()
+
+                # Make sure the digest matches.
+                sha = h.digest()
+                digest_str = info.digest.replace(OBJ_DIGEST_TYPE, "")
```

Comment:
```suggestion
                digest_str = info.digest.replace(OBJ_DIGEST_TYPE, "").replace(OBJ_DIGEST_TYPE.upper(), "")
```

Reply:
Hey, could you elaborate, why somthing like this?

Reply:
I've seen both `"digest": "sha-256=xxx"` and `"digest": "SHA-256=xxx"`. Probably we should change this line instead: https://github.com/tothandras/nats.py/blob/object_store/nats/js/object_store.py#L50. The Go client uses `SHA`: https://github.com/nats-io/nats.go/blob/f4102422df7df83b2032318fd14a248c45f113e1/object.go#L232